File: src/main/java/com/github/jd499/jtrivia/model/Clue.java
package com.github.jd499.jtrivia.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.persistence.*;
import lombok.Data;
import java.time.LocalDateTime;

@Entity
@Table(name = "clues")
@Data
public class Clue {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column(nullable = false, columnDefinition = "text")
    private String answer;

    @Column(nullable = false, columnDefinition = "text")
    private String question;

    private Integer value;

    @Column
    private LocalDateTime airdate;

    @Column(name = "created_at")
    private LocalDateTime createdAt;

    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "category_id")
    @JsonIgnoreProperties({"createdAt", "updatedAt"})
    private Category category;

    @Column(name = "game_id")
    private Integer gameId;

    @Column(name = "invalid_count")
    private Integer invalidCount = 0;

    @Enumerated(EnumType.STRING)
    @Column(name = "round_type", nullable = false)
    private RoundType roundType;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }
}

File: src/main/kotlin/com/jd/jtrivia/models/Clue.kt
package com.jd.jtrivia.models

import jakarta.persistence.*
import java.time.LocalDate
import java.time.LocalDateTime

@Entity
data class Clue(
    @Id val id: Long,
    val answer: String,
    @Column(length = 1000) val question: String,
    val airdate: LocalDate,
    val shown: Boolean = false,
    @ManyToOne(fetch = FetchType.EAGER) @JoinColumn(name = "category_id") val category: Category,
    @Column(name = "`VALUE`") val value: Int? = null,
    val game_id: Int? = null,
    val invalid_count: Int? = null,
    @Column(name = "created_at", updatable = false)
    val createdAt: LocalDateTime = LocalDateTime.now(),
    @Column(name = "updated_at") val updatedAt: LocalDateTime = LocalDateTime.now()
)
